fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -81,6 +83,15 @@
 import static org.apache.hadoop.hive.common.FileUtils.unescapePathName;
 
 public class PartitionUtil {
+    // used for compute date/datetime literal
+    public enum DateInterval {
```

Comment:
`HOUR` is not a `DateInterval`

Reply:
How about IcebergPartitionTransfom directly? according: [partition-transforms](https://iceberg.apache.org/spec/#partition-transforms)